tests/vector_stores/test_pinecone.py

```diff
@@ -0,0 +1,62 @@
+import builtins
+import unittest
+from typing import Any, Callable
+from unittest.mock import patch
+
+from llama_index.vector_stores.pinecone import (
+    PineconeVectorStore,  # Replace with the actual import
+)
+
+
+class MockPineconePods:
+    version = "2.9.9"
+
+    @staticmethod
+    def init(api_key: str, environment: str) -> None:
+        pass
+
+    class Index:
+        def __init__(self, index_name: str) -> None:
+            pass
+
+
+class MockPineconeServerless:
+    version = "3.0.0"
+
+    class Pinecone:
+        def __init__(self, api_key: str) -> None:
+            pass
+
+        class Index:
+            def __init__(self, index_name: str) -> None:
+                pass
+
+
+# Define the mock import function
```

Comment:
Question: do you get IDE / type system errors without these comments present?

`# type: ignore[name-defined]`

Reply:
I do! `mypy` fails without them b/c it doesn't understand the dynamic import stuff we do